File: 724-find-pivot-index/724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left, right = 0, 0
        
        for i in range(1, len(nums)):
            right += nums[i]
            
        pivot = 0
        while pivot < len(nums) and left != right:
            left += nums[pivot]
            pivot += 1
            if pivot < len(nums):
                right -= nums[pivot]
            
        return pivot if pivot < len(nums) else -1

File: 724-find-pivot-index/724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int right = 0;
        for (int i = 0; i < nums.length; i++) {
            right += nums[i];
        }
        
        int left = 0;
        for (int i = 0; i < nums.length; i++) {
            right -= nums[i];
            if (left == right) {
                return i;
            } else {
                left += nums[i];
            }
        }
        return -1;
    }
}